cql3/expr/expression.hh

```diff
@@ -217,6 +219,21 @@ struct untyped_constant {
     sstring raw_text;
 };
 
+// A value serialized with the internal (latest) cql_serialization_format
+struct constant {
+    cql3::raw_value value;
+    data_type type; // Never nullptr, for NULL and UNSET might be empty_type
+
+    constant(cql3::raw_value val, data_type typ);
```

Comment:
I wanted to ensure that each time a `constant` is created, it has both the value and the type.
Without this someone could by mistake write something like:
```c++
constant a{raw_value::make_null()};
```
Which would cause the type to be `nullptr`

Reply:
OK.